### ISSUE TITLE
[BUG] winget's PowerShell module returns updates that WingetUI does not

### ISSUE THREAD

**Issue by Lockszmith-GH** [marticliment/WingetUI#1672]
### Please confirm these before moving forward

- [X] I have searched for my issue and not found a work-in-progress/duplicate/resolved issue.
- [X] I have tested that this issue has not been fixed in the latest [(beta or stable) release](https://github.com/marticliment/WingetUI/releases/).
- [X] I have checked the [FAQ](https://github.com/marticliment/WingetUI#frequently-asked-questions) section for solutions
- [X] This issue is about a bug (if it is not, please use the [correct template](https://github.com/marticliment/WingetUI/issues/new/choose))

### Describe your issue

After I completed all updates with WinGetUI, one of my apps, prompted me for an update, but it wasn't listed as available in WingetUI.

Using the official [Microsoft.WinGet.Client PowerShell Module](https://github.com/microsoft/winget-cli/tree/master/src/PowerShell/Microsoft.WinGet.Client) has become the most convenient cli method for me, so I listed the available updates using powershell and got a list of additional packages not listed in the UI.

Things I treid to validate the issue is WinGetUI and not my actions:
- Reset the winget sources, and refresh them
- After changing to the system winget and restarting, WinGetUI listed some packages, but not all of them.

The Log output below is right at restart, using the internal WinGet when it returned nothing.
Interestingly enough, the **Total packages** entry in the log matches the count from PowerShell: 242.

### Steps to reproduce the issue

Make sure you are using the latest PowerShell (not Windows PowerShell)
```cmd
winget install --id Microsoft.Powershell
```

Launch `pwsh` and install the module:
```PowerShell
Install-Module Microsoft.WinGet.Client
```

Simplest way would be to clean WinGetUI Software Updates view by updating everything, otherwise compare the output below to the view:

```PowerShell
$wglist = Get-WinGetPackage; $wgupd = $wglist | Where-Object { $_.IsUpdateAvailable }
Write-Host "Packages $($wglist.Count); Updates $($wgupd.Count)"
$wgupd
```

### WingetUI Log

<details><summary>WingetUI Log</summary>

```text
🟢 Using cached lang file
🟡 Using bundled lang file (forced=True)
🔵 It took 0.04169011116027832 to load all language files
🟢 checkQueue Thread started!
🔵 Language file up-to-date
🔵 Found default chocolatey installation on expected location
---------------------------------------------------------------------------------------------------

   WingetUI version 2.2.0 (version number 2.2) log
   All modules loaded successfully and sys.stdout patched correctly, starting main script
   Translator function language set to "en"

---------------------------------------------------------------------------------------------------

 Log legend:
 🔵: Verbose
 🟢: Information
 🟡: Warning
 🟠: Handled unexpected exception
 🔴: Unhandled unexpected exception

🔴  Traceback (most recent call last):

🔴  Traceback (most recent call last):

🔴    File ".\__main__.py", line 69, in <module>

🔴    File ".\__main__.py", line 69, in <module>

🔴    File ".\PackageEngine\Classes.py", line 39, in __init__

🔴    File ".\PackageEngine\Classes.py", line 39, in __init__

🔴  AttributeError: 'NoneType' object has no attribute 'NAME'

🔴  AttributeError: 'NoneType' object has no attribute 'NAME'

🔴 Note this traceback was caught by reporter and has been added to the log ('NoneType' object has no attribute 'NAME')
🔵 Starting main application...
🔵 Looking for alive instances...
🔵 Starting API with random session authentication token e9b7762278f502f3145a9f192c39c69a7345d89b9ddfeafe1c8a1b65df3fad21
🟡 Found lock file(s), reactivating...
🔵 Loading WingetUI AUMID...
🔵 Starting update check
🟢 Downloaded latest metadata to local file from url https://raw.githubusercontent.com/marticliment/WingetUI/main/WebBasedData/screenshot-database-v2.json
🔵 Loaded metadata from local file
🔵 Starting Winget source search...
🟢 Winget source search finished with 2 sources
🔵 Reactivation signal ignored: RaiseWindow_[1704030008.678393]
🟢 Found valid aumid {6D809377-6AF0-444B-8957-A3773F02200E}\WingetUI\wingetui.exe
{'sudoFound': True, 'sudoVersion': 'gsudo v2.4.2 (Branch.tags-v2.4.2.Sha.c23b1024b4974f5d90cf82968637bdb4c8924ca7)\r', 'ScoopFound': False, 'ScoopVersion': 'Scoop is disabled', 'ChocolateyFound': False, 'ChocolateyVersion': 'Chocolatey is disabled', 'PipFound': False, 'PipVersion': 'Pip is disabled', 'NpmFound': False, 'NpmVersion': 'Npm is disabled', '.NET ToolFound': False, '.NET ToolVersion': '.NET Tool is disabled', 'PowerShellFound': False, 'PowerShellVersion': 'PowerShell is disabled', 'WingetFound': True, 'WingetVersion': 'v1.6.3482  '}
🔵 Reached main ui load milestone
🟢 Discover tab loaded successfully
🟢 Total packages: 0
🟢 Update tab loaded successfully
🟢 Searching for string ""
🔵 Reloading sources...
🔵 Reloading Winget sources...
🟢 Uninstall tab loaded successfully
🔵 Starting Winget search for installed packages
🟢 Searching for string ""
🔵 Reloading Scoop sources...
🔵 Starting Winget source search...
🔵 Starting Scoop source search...
🔵 Starting Chocolatey source search...
🟢 Winget source search finished with 2 sources
🔵 Starting PowerShell source search...
🟢 Settings tab loaded!
🟢 About tab loaded!
🟢 Main application loaded...
🟢 Total packages: 0
🟢 Chocolatey source search finished with 1 sources
Traceback (most recent call last):
  File ".\Interface\CustomWidgets\InstallerWidgets.py", line 669, in <lambda>
TypeError: 'NoneType' object is not callable
Traceback (most recent call last):
  File ".\Interface\CustomWidgets\InstallerWidgets.py", line 669, in <lambda>
TypeError: 'NoneType' object is not callable
Traceback (most recent call last):
  File ".\Interface\CustomWidgets\InstallerWidgets.py", line 669, in <lambda>
TypeError: 'NoneType' object is not callable
🟢 Scoop source search finished with 0 sources
🔵 Starting Winget search for updates
🟡 package Update for  (KB2504637) failed parsing, going for method 2...
🟢 Winget search for installed packages finished with 242 result(s)
🟢 Searching for string ""
🟢 Searching for string ""
🟢 Total packages: 242
🔵 Starting Winget source search...
🔵 Starting Scoop source search...
🔵 Starting Chocolatey source search...
Traceback (most recent call last):
  File ".\Interface\CustomWidgets\InstallerWidgets.py", line 669, in <lambda>
TypeError: 'NoneType' object is not callable
Traceback (most recent call last):
  File ".\Interface\CustomWidgets\InstallerWidgets.py", line 669, in <lambda>
TypeError: 'NoneType' object is not callable
🔵 Announcement URL: https://www.marticliment.com/resources/wingetui.announcement
🔵 Image URL: https://www.marticliment.com/resources/wingetui.png
🟢 Winget source search finished with 2 sources
🟢 Scoop source search finished with 4 sources
🟢 Chocolatey source search finished with 1 sources
🟢 Winget search for updates finished with 5 result(s)
🟡 Multiple versions of ImageMagick.ImageMagick are installed, latest version is installed. Not showing the update
🟡 Package XnSoft.XnViewMP has version 1.6.3.0 ignored
🟡 Package PeterDaveHello.TransmissionRemoteGUI has version 5.18.0 ignored
🟡 Package Microsoft.VCRedist.2013.x86 has version 12.0.40664.0 ignored
🟡 Package Microsoft.VCRedist.2013.x86 has version 12.0.40664.0 ignored
🟢 Searching for string ""
🟢 Searching for string ""
🟢 Total packages: 0
🟢 PowerShell source search finished with 1 sources
🔵 Announcement URL: https://www.marticliment.com/resources/wingetui.announcement
🔵 Image URL: https://www.marticliment.com/resources/wingetui.png
HTTP Error 524: 
🔵 Version URL: https://versions.marticliment.com/versions/wingetui.ver
🟢 Updates not found
🔵 Loading WingetUI log...
🔵 Copying log to the clipboard...
🔵 Loading WingetUI log...
```

</details>

### Package Managers Logs

```text
N/A
```

### Screenshots and videos

![image](https://github.com/marticliment/WingetUI/assets/905716/6a93b51b-f340-4781-875a-8a9352bf2f72)

**Comment by marticliment** [comment]
Hello,

does C:\Program Files\WingetUI\PackageEngine\Managers\winget-cli_x64\winget.exe report those missing updates?

**Comment by Lockszmith-GH** [comment]
Almost, it reports something, but not everything:

![image](https://github.com/marticliment/WingetUI/assets/905716/4c554e28-397a-4e8e-bc37-352670104fee)

I've tried with --include-unknown, and it's a bit different, but still doesn't match PowerShell output.

Then, right before I hit send here, I tried:
```PowerShell
& "${env:LOCALAPPDATA}\Microsoft\WindowsApps\winget.exe" upgrade --include-unknown
```

and got the 12 packages split into 2 sections.

So it seems to be a mixture of the winget version, as well as the --include-unknown flag.

**Comment by marticliment** [comment]
I will study this.

**Comment by Lockszmith-GH** [comment]
Thanks, it was really late yesterday, and so I didn't go through better collection, below are some additional output:

<details><summary>Output of: <code>& "${env:LOCALAPPDATA}\Microsoft\WindowsApps\winget.exe" upgrade --include-unknown</code></summary>

```log
Name                                                         Id                                   Version      Available    Source
----------------------------------------------------------------------------------------------------------------------------------
ImageMagick 7.1.0-62 Q16-HDRI (64-bit) (2023-02-12)          ImageMagick.ImageMagick              7.1.0.62     7.1.1.24     winget
RawTherapee version 5.8                                      RawTherapee.RawTherapee              5.8          5.9          winget
Voicemeeter, The Virtual Mixing Console                      VB-Audio.Voicemeeter.Banana          Unknown      2.0.6.8      winget
XnViewMP 1.5.5                                               XnSoft.XnViewMP                      1.5.5        1.6.3.0      winget
Transmission Remote GUI 5.18                                 PeterDaveHello.TransmissionRemoteGUI Unknown      5.18.0       winget
NAPS2                                                        Cyanfish.NAPS2                       6.1.2        7.2.2        winget
Microsoft Visual C++ 2013 Redistributable (x86) - 12.0.40660 Microsoft.VCRedist.2013.x86          12.0.40660.0 12.0.40664.0 winget
Python 3.12.1 (64-bit)                                       Python.Python.3.10                   > 3.10.11    3.12.1       winget
Microsoft Visual C++ 2013 Redistributable (x86) - 12.0.21005 Microsoft.VCRedist.2013.x86          12.0.21005.1 12.0.40664.0 winget
Microsoft Windows Desktop Runtime - 6.0.25 (x64)             Microsoft.DotNet.DesktopRuntime.7    < 7.0.0      7.0.14       winget
11 upgrades available.

The following packages have an upgrade available, but require explicit targeting for upgrade:
Name           Id              Version       Available            Source
------------------------------------------------------------------------
Microsoft Edge Microsoft.Teams 120.0.2210.91 23257.2618.2432.4374 winget
```

</details>

<details><summary>Output of: <code>& "${env:LOCALAPPDATA}\Microsoft\WindowsApps\winget.exe" upgrade</code></summary>

```log
Name                                                         Id                                Version      Available    Source
-------------------------------------------------------------------------------------------------------------------------------
ImageMagick 7.1.0-62 Q16-HDRI (64-bit) (2023-02-12)          ImageMagick.ImageMagick           7.1.0.62     7.1.1.24     winget
RawTherapee version 5.8                                      RawTherapee.RawTherapee           5.8          5.9          winget
XnViewMP 1.5.5                                               XnSoft.XnViewMP                   1.5.5        1.6.3.0      winget
NAPS2                                                        Cyanfish.NAPS2                    6.1.2        7.2.2        winget
Microsoft Visual C++ 2013 Redistributable (x86) - 12.0.40660 Microsoft.VCRedist.2013.x86       12.0.40660.0 12.0.40664.0 winget
Python 3.12.1 (64-bit)                                       Python.Python.3.10                > 3.10.11    3.12.1       winget
Microsoft Visual C++ 2013 Redistributable (x86) - 12.0.21005 Microsoft.VCRedist.2013.x86       12.0.21005.1 12.0.40664.0 winget
Microsoft Windows Desktop Runtime - 6.0.25 (x64)             Microsoft.DotNet.DesktopRuntime.7 < 7.0.0      7.0.14       winget
9 upgrades available.

The following packages have an upgrade available, but require explicit targeting for upgrade:
Name           Id              Version       Available            Source
------------------------------------------------------------------------
Microsoft Edge Microsoft.Teams 120.0.2210.91 23257.2618.2432.4374 winget
2 package(s) have version numbers that cannot be determined. Use --include-unknown to see all results.
```

</details>

<details><summary>Output of: <code>& "C:\Program Files\WingetUI\wingetui\PackageEngine\Managers\winget-cli_x64\winget.exe" upgrade --include-unknown</code></summary>

```log
Name                                                         Id                                   Version      Available    Source
----------------------------------------------------------------------------------------------------------------------------------
ImageMagick 7.1.0-62 Q16-HDRI (64-bit) (2023-02-12)          ImageMagick.ImageMagick              7.1.0.62     7.1.1.24     winget
XnViewMP 1.5.5                                               XnSoft.XnViewMP                      1.5.5        1.6.3.0      winget
Transmission Remote GUI 5.18                                 PeterDaveHello.TransmissionRemoteGUI Unknown      5.18.0       winget
Microsoft Visual C++ 2013 Redistributable (x86) - 12.0.40660 Microsoft.VCRedist.2013.x86          12.0.40660.0 12.0.40664.0 winget
Microsoft Visual C++ 2013 Redistributable (x86) - 12.0.21005 Microsoft.VCRedist.2013.x86          12.0.21005.1 12.0.40664.0 winget
5 upgrades available.
```

</details>

<details><summary>Output of: <code>& "C:\Program Files\WingetUI\PackageManagers\winget-cli_x64\winget.exe" upgrade --include-unknown</code> (same as above)</summary>

```log
Name                                                         Id                                   Version      Available    Source
----------------------------------------------------------------------------------------------------------------------------------
ImageMagick 7.1.0-62 Q16-HDRI (64-bit) (2023-02-12)          ImageMagick.ImageMagick              7.1.0.62     7.1.1.24     winget
XnViewMP 1.5.5                                               XnSoft.XnViewMP                      1.5.5        1.6.3.0      winget
Transmission Remote GUI 5.18                                 PeterDaveHello.TransmissionRemoteGUI Unknown      5.18.0       winget
Microsoft Visual C++ 2013 Redistributable (x86) - 12.0.40660 Microsoft.VCRedist.2013.x86          12.0.40660.0 12.0.40664.0 winget
Microsoft Visual C++ 2013 Redistributable (x86) - 12.0.21005 Microsoft.VCRedist.2013.x86          12.0.21005.1 12.0.40664.0 winget
5 upgrades available.
```

</details>

<details><summary>Output of: <code>& "C:\Program Files\WingetUI\wingetui\PackageEngine\Managers\winget-cli_x64\winget.exe" upgrade</code></summary>

```log
Name                                                         Id                          Version      Available    Source
-------------------------------------------------------------------------------------------------------------------------
ImageMagick 7.1.0-62 Q16-HDRI (64-bit) (2023-02-12)          ImageMagick.ImageMagick     7.1.0.62     7.1.1.24     winget
XnViewMP 1.5.5                                               XnSoft.XnViewMP             1.5.5        1.6.3.0      winget
Microsoft Visual C++ 2013 Redistributable (x86) - 12.0.40660 Microsoft.VCRedist.2013.x86 12.0.40660.0 12.0.40664.0 winget
Microsoft Visual C++ 2013 Redistributable (x86) - 12.0.21005 Microsoft.VCRedist.2013.x86 12.0.21005.1 12.0.40664.0 winget
4 upgrades available.
1 package(s) have version numbers that cannot be determined. Use --include-unknown to see all results.
```

</details>

**Comment by marticliment** [comment]
Hello, What does "${env:LOCALAPPDATA}\Microsoft\WindowsApps\winget.exe" --version report?

**Comment by Lockszmith-GH** [comment]
```log
> & "${env:LOCALAPPDATA}\Microsoft\WindowsApps\winget.exe" --version
v1.6.3482

> & "C:\Program Files\WingetUI\PackageManagers\winget-cli_x64\winget.exe" --version
v1.6.2771
```

**Comment by Lockszmith-GH** [comment]
I reinstalled WinGetUI, now I only have a single cli_x64 winget.exe, this is the output:

```
> & "${env:LOCALAPPDATA}\Microsoft\WindowsApps\winget.exe" --version
v1.6.3482

> & "C:\Program Files\WingetUI\wingetui\PackageEngine\Managers\winget-cli_x64\winget.exe" --version
v1.6.3482
```

However, the output is still the same as reported above.
I've edited out the old winget.exe output above - probably a relic of an older version.

**Comment by Lockszmith-GH** [comment]
I reinstalled WinGetUI, now I only have a single cli_x64 winget.exe, this is the output:

```
> & "${env:LOCALAPPDATA}\Microsoft\WindowsApps\winget.exe" --version
v1.6.3482

> & "C:\Program Files\WingetUI\wingetui\PackageEngine\Managers\winget-cli_x64\winget.exe" --version
v1.6.3482
```

I've made sure Ignored Updates are clear, and now I have the following in the UI:
![image](https://github.com/marticliment/WingetUI/assets/905716/835544fc-aca1-4344-a84d-fe3b6d063680)

<details><summary>WingetUI Log on startup</summary>

```log
🟢 Using cached lang file
🟡 Using bundled lang file (forced=True)
🔵 It took 0.004007101058959961 to load all language files
🟢 checkQueue Thread started!
🔵 Language file up-to-date
🔵 Found default chocolatey installation on expected location
---------------------------------------------------------------------------------------------------

   WingetUI version 2.2.0 (version number 2.2) log
   All modules loaded successfully and sys.stdout patched correctly, starting main script
   Translator function language set to "en"

---------------------------------------------------------------------------------------------------

 Log legend:
 🔵: Verbose
 🟢: Information
 🟡: Warning
 🟠: Handled unexpected exception
 🔴: Unhandled unexpected exception

🔵 Starting main application...
🔵 Looking for alive instances...
🔵 Starting API with random session authentication token 42f3ecc12761f13e0dc293d5f95bb04ede4462869eb8ee22e02506fdd512bc7b
🟡 Found lock file(s), reactivating...
🔵 Reloading Winget sources...
🔵 Loading WingetUI AUMID...
🔵 Starting update check
🔵 Version URL: https://www.marticliment.com/versions/wingetui.ver
🟢 Updates not found
🟢 Downloaded latest metadata to local file from url https://raw.githubusercontent.com/marticliment/WingetUI/main/WebBasedData/screenshot-database-v2.json
🔵 Loaded metadata from local file
🔵 Starting Winget source search...
🟢 Winget source search finished with 2 sources
🔵 Reactivation signal ignored: RaiseWindow_[1704217255.4511662]
🟢 Found valid aumid {6D809377-6AF0-444B-8957-A3773F02200E}\WingetUI\wingetui.exe
{'sudoFound': True, 'sudoVersion': 'gsudo v2.0.4 (Branch.tags-v2.0.4.Sha.506efa024af0cef6e4b0cfec42e0c8c5d0b1472c)\r', 'ScoopFound': False, 'ScoopVersion': 'Scoop is disabled', 'ChocolateyFound': False, 'ChocolateyVersion': 'Chocolatey is disabled', 'PipFound': False, 'PipVersion': 'Pip is disabled', 'NpmFound': False, 'NpmVersion': 'Npm is disabled', '.NET ToolFound': False, '.NET ToolVersion': '.NET Tool is disabled', 'PowerShellFound': False, 'PowerShellVersion': 'PowerShell is disabled', 'WingetFound': True, 'WingetVersion': 'v1.6.3482  '}
🔵 Reached main ui load milestone
🟢 Discover tab loaded successfully
🟢 Total packages: 0
🟢 Update tab loaded successfully
🟢 Searching for string ""
🔵 Reloading sources...
🔵 Reloading Winget sources...
🟢 Uninstall tab loaded successfully
🔵 Starting Winget search for installed packages
🟢 Searching for string ""
🔵 Reloading Scoop sources...
🔵 Starting Winget source search...
🔵 Starting Scoop source search...
🔵 Starting Chocolatey source search...
🔵 Starting PowerShell source search...
🟢 Settings tab loaded!
🟢 Winget source search finished with 2 sources
🟢 About tab loaded!
🟢 Scoop source search finished with 0 sources
🟢 Chocolatey source search finished with 0 sources
🟢 Main application loaded...
🟢 Total packages: 0
Traceback (most recent call last):
  File ".\Interface\CustomWidgets\InstallerWidgets.py", line 669, in <lambda>
TypeError: 'NoneType' object is not callable
Traceback (most recent call last):
  File ".\Interface\CustomWidgets\InstallerWidgets.py", line 669, in <lambda>
TypeError: 'NoneType' object is not callable
Traceback (most recent call last):
  File ".\Interface\CustomWidgets\InstallerWidgets.py", line 669, in <lambda>
TypeError: 'NoneType' object is not callable
🔵 Starting Winget search for updates
🟡 package Update for  (KB2504637) failed parsing, going for method 2...
🟢 Winget search for installed packages finished with 242 result(s)
🟢 Searching for string ""
🟢 Searching for string ""
🟢 Total packages: 242
🟢 Winget search for updates finished with 5 result(s)
🟡 Multiple versions of ImageMagick.ImageMagick are installed, latest version is installed. Not showing the update
🟢 Searching for string ""
Exception ignored in: <function BaseWindowsToaster.show_toast.<locals>.<lambda> at 0x000001CD0324B420>
Traceback (most recent call last):
  File "windows_toasts\toasters.py", line 147, in <lambda>
🟢 Searching for string ""
  File ".\Interface\CustomWidgets\GenericWidgets.py", line 738, in <lambda>
TypeError: ToastNotification.reportException() missing 1 required positional argument: 'id'
🟡 Can't get custom interval time! (got value was '')
🟢 Total packages: 4
🟢 PowerShell source search finished with 1 sources
🔵 Loading WingetUI log...
🔵 Copying log to the clipboard...
🔵 Loading WingetUI log...

```

</details>

Below is the output of the cli tools, system and the WinGetUI version

<details><summary>Output of: <code>& "C:\Program Files\WingetUI\wingetui\PackageEngine\Managers\winget-cli_x64\winget.exe" upgrade</code></summary>

```log
Name                                                         Id                          Version      Available    Source
-------------------------------------------------------------------------------------------------------------------------
ImageMagick 7.1.0-62 Q16-HDRI (64-bit) (2023-02-12)          ImageMagick.ImageMagick     7.1.0.62     7.1.1.24     winget
XnViewMP 1.5.5                                               XnSoft.XnViewMP             1.5.5        1.6.3.0      winget
Microsoft Visual C++ 2013 Redistributable (x86) - 12.0.40660 Microsoft.VCRedist.2013.x86 12.0.40660.0 12.0.40664.0 winget
Microsoft Visual C++ 2013 Redistributable (x86) - 12.0.21005 Microsoft.VCRedist.2013.x86 12.0.21005.1 12.0.40664.0 winget
4 upgrades available.
1 package(s) have version numbers that cannot be determined. Use --include-unknown to see all results.
```

</details>

<details><summary>Output of: <code>& "C:\Program Files\WingetUI\wingetui\PackageEngine\Managers\winget-cli_x64\winget.exe" upgrade --include-unknown
</code></summary>

```log
Name                                                         Id                                   Version      Available    Source
----------------------------------------------------------------------------------------------------------------------------------
ImageMagick 7.1.0-62 Q16-HDRI (64-bit) (2023-02-12)          ImageMagick.ImageMagick              7.1.0.62     7.1.1.24     winget
XnViewMP 1.5.5                                               XnSoft.XnViewMP                      1.5.5        1.6.3.0      winget
Transmission Remote GUI 5.18                                 PeterDaveHello.TransmissionRemoteGUI Unknown      5.18.0       winget
Microsoft Visual C++ 2013 Redistributable (x86) - 12.0.40660 Microsoft.VCRedist.2013.x86          12.0.40660.0 12.0.40664.0 winget
Microsoft Visual C++ 2013 Redistributable (x86) - 12.0.21005 Microsoft.VCRedist.2013.x86          12.0.21005.1 12.0.40664.0 winget
5 upgrades available.
```

</details>

The following is a screenshot when using the System Winget under advanced settings:
![image](https://github.com/marticliment/WingetUI/assets/905716/da1798f1-4dcb-4004-9363-1544d17c3a2b)

<details><summary>WingetUI Log on startup</summary>

```log
🟢 Using cached lang file
🟡 Using bundled lang file (forced=True)
🔵 It took 0.0040051937103271484 to load all language files
🟢 checkQueue Thread started!
🔵 Language file up-to-date
🔵 Found default chocolatey installation on expected location
---------------------------------------------------------------------------------------------------

   WingetUI version 2.2.0 (version number 2.2) log
   All modules loaded successfully and sys.stdout patched correctly, starting main script
   Translator function language set to "en"

---------------------------------------------------------------------------------------------------

 Log legend:
 🔵: Verbose
 🟢: Information
 🟡: Warning
 🟠: Handled unexpected exception
 🔴: Unhandled unexpected exception

🔵 Starting main application...
🔵 Looking for alive instances...
🔵 Starting API with random session authentication token d04fc8641130c4f31c4d324836a00c562e7fe032339d4f785cd20b6b2aed45a0
🟡 Found lock file(s), reactivating...
🔵 Reloading Winget sources...
🔵 Loading WingetUI AUMID...
🔵 Starting update check
🔵 Version URL: https://www.marticliment.com/versions/wingetui.ver
🟢 Updates not found
🟢 Downloaded latest metadata to local file from url https://raw.githubusercontent.com/marticliment/WingetUI/main/WebBasedData/screenshot-database-v2.json
🔵 Loaded metadata from local file
🔵 Starting Winget source search...
🟢 Winget source search finished with 2 sources
🔵 Reactivation signal ignored: RaiseWindow_[1704215957.8059437]
🟢 Found valid aumid {6D809377-6AF0-444B-8957-A3773F02200E}\WingetUI\wingetui.exe
{'sudoFound': True, 'sudoVersion': 'gsudo v2.0.4 (Branch.tags-v2.0.4.Sha.506efa024af0cef6e4b0cfec42e0c8c5d0b1472c)\r', 'ScoopFound': False, 'ScoopVersion': 'Scoop is disabled', 'ChocolateyFound': False, 'ChocolateyVersion': 'Chocolatey is disabled', 'PipFound': False, 'PipVersion': 'Pip is disabled', 'NpmFound': False, 'NpmVersion': 'Npm is disabled', '.NET ToolFound': False, '.NET ToolVersion': '.NET Tool is disabled', 'PowerShellFound': False, 'PowerShellVersion': 'PowerShell is disabled', 'WingetFound': True, 'WingetVersion': 'v1.6.3482  '}
🔵 Reached main ui load milestone
🟢 Discover tab loaded successfully
🟢 Total packages: 0
🟢 Update tab loaded successfully
🟢 Searching for string ""
🔵 Reloading sources...
🔵 Reloading Winget sources...
🟢 Uninstall tab loaded successfully
🔵 Starting Winget search for installed packages
🟢 Searching for string ""
🔵 Reloading Scoop sources...
🔵 Starting Winget source search...
🔵 Starting Scoop source search...
🔵 Starting Chocolatey source search...
🔵 Starting PowerShell source search...
🟢 Settings tab loaded!
🟢 Winget source search finished with 2 sources
🟢 About tab loaded!
🟢 Chocolatey source search finished with 1 sources
🟢 Scoop source search finished with 4 sources
🟢 Main application loaded...
🟢 Total packages: 0
Traceback (most recent call last):
  File ".\Interface\CustomWidgets\InstallerWidgets.py", line 669, in <lambda>
TypeError: 'NoneType' object is not callable
Traceback (most recent call last):
  File ".\Interface\CustomWidgets\InstallerWidgets.py", line 669, in <lambda>
TypeError: 'NoneType' object is not callable
Traceback (most recent call last):
  File ".\Interface\CustomWidgets\InstallerWidgets.py", line 669, in <lambda>
TypeError: 'NoneType' object is not callable
🟡 package Update for  (KB2504637) failed parsing, going for method 2...
🟢 Winget search for installed packages finished with 242 result(s)
🟢 Searching for string ""
🟢 Searching for string ""
🟢 Total packages: 242
🔵 Starting Winget search for updates
🟢 Winget search for updates finished with 116 result(s)
🟡 Multiple versions of ImageMagick.ImageMagick are installed, latest version is installed. Not showing the update
🟢 Searching for string ""
Exception ignored in: <function BaseWindowsToaster.show_toast.<locals>.<lambda> at 0x000002BE6E151EE0>
Traceback (most recent call last):
  File "windows_toasts\toasters.py", line 147, in <lambda>
🟢 Searching for string ""
  File ".\Interface\CustomWidgets\GenericWidgets.py", line 738, in <lambda>
🟡 Can't get custom interval time! (got value was '')
TypeError: ToastNotification.reportException() missing 1 required positional argument: 'id'
🟢 Total packages: 10
🟢 PowerShell source search finished with 1 sources
🔵 Starting update check
🔵 Version URL: https://www.marticliment.com/versions/wingetui.ver
🟢 Updates not found
🔵 Loading WingetUI log...
🔵 Copying log to the clipboard...
🔵 Loading WingetUI log...
```

</details>

<details><summary>Output of: <code>& "${env:LOCALAPPDATA}\Microsoft\WindowsApps\winget.exe" upgrade</code></summary>

```log
Name                                                         Id                                Version      Available    Source
-------------------------------------------------------------------------------------------------------------------------------
ImageMagick 7.1.0-62 Q16-HDRI (64-bit) (2023-02-12)          ImageMagick.ImageMagick           7.1.0.62     7.1.1.24     winget
RawTherapee version 5.8                                      RawTherapee.RawTherapee           5.8          5.9          winget
XnViewMP 1.5.5                                               XnSoft.XnViewMP                   1.5.5        1.6.3.0      winget
NAPS2                                                        Cyanfish.NAPS2                    6.1.2        7.2.2        winget
Microsoft Visual C++ 2013 Redistributable (x86) - 12.0.40660 Microsoft.VCRedist.2013.x86       12.0.40660.0 12.0.40664.0 winget
Python 3.12.1 (64-bit)                                       Python.Python.3.10                > 3.10.11    3.12.1       winget
Microsoft Visual C++ 2013 Redistributable (x86) - 12.0.21005 Microsoft.VCRedist.2013.x86       12.0.21005.1 12.0.40664.0 winget
Microsoft Windows Desktop Runtime - 6.0.25 (x64)             Microsoft.DotNet.DesktopRuntime.7 < 7.0.0      7.0.14       winget
9 upgrades available.

The following packages have an upgrade available, but require explicit targeting for upgrade:
Name           Id              Version       Available            Source
------------------------------------------------------------------------
Microsoft Edge Microsoft.Teams 120.0.2210.91 23257.2618.2432.4374 winget
2 package(s) have version numbers that cannot be determined. Use --include-unknown to see all results.
```

</details>

<details><summary>Output of: <code>& "${env:LOCALAPPDATA}\Microsoft\WindowsApps\winget.exe" upgrade --include-unknown</code></summary>

```log
Name                                                         Id                                   Version      Available    Source
----------------------------------------------------------------------------------------------------------------------------------
ImageMagick 7.1.0-62 Q16-HDRI (64-bit) (2023-02-12)          ImageMagick.ImageMagick              7.1.0.62     7.1.1.24     winget
RawTherapee version 5.8                                      RawTherapee.RawTherapee              5.8          5.9          winget
Voicemeeter, The Virtual Mixing Console                      VB-Audio.Voicemeeter.Banana          Unknown      2.0.6.8      winget
XnViewMP 1.5.5                                               XnSoft.XnViewMP                      1.5.5        1.6.3.0      winget
Transmission Remote GUI 5.18                                 PeterDaveHello.TransmissionRemoteGUI Unknown      5.18.0       winget
NAPS2                                                        Cyanfish.NAPS2                       6.1.2        7.2.2        winget
Microsoft Visual C++ 2013 Redistributable (x86) - 12.0.40660 Microsoft.VCRedist.2013.x86          12.0.40660.0 12.0.40664.0 winget
Python 3.12.1 (64-bit)                                       Python.Python.3.10                   > 3.10.11    3.12.1       winget
Microsoft Visual C++ 2013 Redistributable (x86) - 12.0.21005 Microsoft.VCRedist.2013.x86          12.0.21005.1 12.0.40664.0 winget
Microsoft Windows Desktop Runtime - 6.0.25 (x64)             Microsoft.DotNet.DesktopRuntime.7    < 7.0.0      7.0.14       winget
11 upgrades available.

The following packages have an upgrade available, but require explicit targeting for upgrade:
Name           Id              Version       Available            Source
------------------------------------------------------------------------
Microsoft Edge Microsoft.Teams 120.0.2210.91 23257.2618.2432.4374 winget
```
</details>

# In conclusion
There is a difference between the embedded winget and the system winget, even though they are reporting the same version.

This might not be considered a bug, and if this was the case I would probably not have reported this.

Whatever was affecting my system, was some relic from previous installation.
Sorry to have wasted your time - I should have done this uninstall/reinstall first, following <details><summary>the first rule of IT</summary>![53bb3f41d1fb17aa28b506c6b03980bd-2047533158](https://github.com/marticliment/WingetUI/assets/905716/b844aaec-745c-49d2-8106-099ddff39065)</details>

## Should I open a  new bug/issue?

**ImageMagick** is still not shown in either, so there is still a minor bug if you still want to pursue this.

**Comment by marticliment** [comment]
> Sorry to have wasted your time - I should have done this uninstall/reinstall first, following

Don't worry, this bug shouldn't have occurred even if the older winget version was there.

I will investigate ImageMagick, since it should be shown on WingetUI.

**Comment by ExpPhysU** [comment]
I am sorry, but winget on windows 11 does not work in wingetui. If I use the above method I get in the latest PowerShell

> & "${env:LOCALAPPDATA}\Microsoft\WindowsApps\winget.exe" --version
v1.6.3482
> & "C:\Program Files\WingetUI\wingetui\PackageEngine\Managers\winget-cli_x64\winget.exe" --version
NO OUTPUT whatsoever 

Somethging does not work. Uninstalling both and reinstalling them does not help.

It would be great if that bug could be found

Thanks.

**Comment by marticliment** [comment]
Please try enabling "Use system Winget" from WingetUI Settings -> Experimental Features
